### PR TITLE
New Extension: acrtransfer - Supporting the Azure Container Registry Transfer feature

### PR DESCRIPTION
**Pull request by mabenedi**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required) **Yes**
- [x] Have you run `python scripts/ci/test_index.py -q` locally? **Yes**

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md). **Yes**

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
acr

**Comment by yungezz** [comment]
hi @mabenedi could you pls share context that these commands need to be a standalone extension, instead of part of core acr command module? thanks. cc @yugangw-msft .

**Comment by mabenedi** [comment]
> hi @mabenedi could you pls share context that these commands need to be a standalone extension, instead of part of core acr command module? thanks. cc @yugangw-msft .

Sure. The Transfer is used exclusively to get container images into airgapped clouds (USNAT and USSEC). Because customers in those clouds are often running very old versions of CLI, we wanted to be able to have an extension so that they could install it immediately directly from a public storage account container. 

Additionally, the number of customers who have access to these clouds is very small and the Transfer feature can get confusingly close to other commands (import vs import-pipeline), so we'd prefer not to ship it to everyone at this time and create confusion/antipatterns.

**Comment by yugangw-msft** [comment]
@yungezz , I agree with @mabenedi's comment which explains the context we stay with extension for a while, as it is a pretty niche feature used by limited number of users till today. We can get into core once user base expends. Please let me know if more information is needed from us

File: src/acrtransfer/azext_acrtransfer/commands.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        min_api='2019-12-01-preview'
+    )
+
+    pipelinerun_sdk = CliCommandType(
+        operations_tmpl='azext_acrtransfer.vendored_sdks.containerregistry.v2019_12_01_preview.operations#PipelineRunsOperations.{}',
+        client_factory=cf_acrtransfer,
+        min_api='2019-12-01-preview'
+    )
+
+    with self.command_group('acr import-pipeline', importpipeline_sdk, table_transformer=import_pipeline_output_format, is_preview=True) as g:
```
Comment (fengzhou-msft) [inline src/acrtransfer/azext_acrtransfer/commands.py:31]: Can you use a subgroup `acr pipeline` and then you have `import/export/run` commands under this subgroup?

File: src/acrtransfer/azext_acrtransfer/commands.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        g.custom_command('create', 'create_importpipeline')
+        g.custom_command('delete', 'delete_importpipeline')
+        g.custom_command('list', 'list_importpipeline')
+        g.custom_command('show', 'get_importpipeline')
```
Comment (fengzhou-msft) [inline src/acrtransfer/azext_acrtransfer/commands.py:35]: Use `custom_show_command` for all show commands as it returns exit code 3 for 404 case.

Comment (mabenedi) [reply src/acrtransfer/azext_acrtransfer/commands.py]: Done

File: src/acrtransfer/azext_acrtransfer/commands.py (hunk 1 above)
Comment (mabenedi) [reply src/acrtransfer/azext_acrtransfer/commands.py:31]: I could, although I feel like it's more user-friendly as it is. Would it be okay to leave it the way it is or should we definitely change it?

Comment (fengzhou-msft) [reply src/acrtransfer/azext_acrtransfer/commands.py:31]: I realized `import-pipeline` and `export-pipeline` are 2 types of pipelines, it's OK to leave it as it is. It's just that the pipeline subgroups (`import-pipeline`, `export-pipeline`, `pipeline-run`) will not be shown together in alphabetic order when users run `az acr -h`.

Comment (mabenedi) [reply src/acrtransfer/azext_acrtransfer/commands.py:31]: Gotcha. Yeah that's fine with me, I'd like to keep it as-is

**Review by fengzhou-msft — APPROVED** [review]

File: src/acrtransfer/azext_acrtransfer/_client_factory.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+
+def cf_acrtransfer(cli_ctx, *_):
```
Comment (fengzhou-msft) [inline src/acrtransfer/azext_acrtransfer/_client_factory.py:7]: Expected 2 blank lines.

Comment (mabenedi) [reply src/acrtransfer/azext_acrtransfer/_client_factory.py]: Added another blank line

File: src/acrtransfer/setup.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+with open('HISTORY.rst', 'r', encoding='utf-8') as f:
+    HISTORY = f.read()
+
+setup(
+    name='acrtransfer',
+    version=VERSION,
+    description='Microsoft Azure Command-Line Tools Acrtransfer Extension',
+    author='Microsoft Corporation',
+    author_email='KraterDev@microsoft.com',
+    url='https://github.com/Azure/azure-cli-extensions',
```
Comment (fengzhou-msft) [inline src/acrtransfer/setup.py:46]:
```suggestion
    url='https://github.com/Azure/azure-cli-extensions/tree/master/src/acrtransfer',
```

File: src/acrtransfer/setup.cfg [hunk 5]
```diff
@@ -0,0 +1,2 @@
+[bdist_wheel]
+universal=1
```
Comment (fengzhou-msft) [inline src/acrtransfer/setup.cfg:2]: Remove as we dropped support for Python 2.

**Pull request merged**